hookman/hookman_generator.py

```diff
@@ -711,6 +711,41 @@ def _generate_windows_body(hooks):
         f"    }}",
         f"",
         f"",
+        f"    class PathGuard {{",
+        f"    public:",
+        f"        PathGuard(std::wstring w_filename)",
+        f"            : path_env{{ get_path() }}",
+        f"        {{",
+        fr'            std::wstring::size_type dir_name_size = w_filename.find_last_of(L"/\\");',
+        f'            std::wstring new_path_env = path_env + L";" + w_filename.substr(0, dir_name_size);',
+        f'            _wputenv_s(L"PATH", new_path_env.data());',
```

Comment:
If it's on C++98, there's no guarantee that `.data()` returns a NUL-terminated string, and it may cause a nasty segfault or it will get an invalid string (this is fixed in C++11 and later, but we don't have guarantees on how the user will compile a plugin). For safety, it's better to change all `.data()` to `.c_str()`.

Reply:
Good suggestion. Thanks!